app/static/js/initialize_map.js

```diff
@@ -0,0 +1,155 @@
+geocoder = new google.maps.Geocoder();
+address = document.getElementById("address");
+//Initial map center coordinates
+INITIAL_CENTER_LAT = 39.952;
+INITIAL_CENTER_LONG = -75.195;
+initial_coords = new google.maps.LatLng(INITIAL_CENTER_LAT, INITIAL_CENTER_LONG);
+//Initial map zoom value
+ZOOM = 17;
+//An integer for the offset between the actual window size and the
+//JavaScript-determined Window size in order to size the map
+WINDOW_OFFSET = 70
+
+//Bounds for the Date Slider. End date is current date.
+var BOUNDS_MIN = new Date(2015, 0, 1);
+var BOUNDS_MAX = new Date();
+
+//Get Incident Report information through HTML and Jinja2 and add markers
+//to the map.
+function addMarkers(map) {
+    var vehicle_ids_str = $("#vehicle_ids").data();
+    var license_plates_str = $("#license_plates").data();
+    var latitudes = $("#latitudes").data();
+    var longitudes = $("#longitudes").data();
+    var dates_str = $("#dates").data();
+    var durations_str = $("#durations").data();
+    var agencies_str = $("#agencies").data();
+    var pictures_str = $("#pictures").data();
+    var descriptions_str = $("#descriptions").data();
+    //Tokenize the strings representing arrays that are given through HTML
+    //by Jinja2
+    var vehicle_ids = (vehicle_ids_str.name).split('\'');
+    var license_plates = (license_plates_str.name).split('\'');
+    var dates = (dates_str.name).split('\'');
+    var durations = (durations_str.name).split('\'');
+    var agencies = (agencies_str.name).split('\'');
+    var pictures = (pictures_str.name).split('\'');
+    var descriptions = (descriptions_str.name).split('\'');
+    for (i = 0; i < (latitudes.name).length; i = i + 1) {
+        //In order that the same marker isn't modified every time,
+        //we create a new function here to create and design each marker
+        $(function() {
+            var marker = new google.maps.Marker({
+                //Use latitude and longitude values from incident report
+                //to set position of marker
+                position:{lat: latitudes.name[i], lng: longitudes.name[i]}
+            });
+
+            //Tie marker to map passed as an argument
+            
+            marker.setMap(map);
+            
+            //Information presented when marker is clicked
+            var contentString = '<div id="content">' +
+                '<div id="siteNotice">' +
+                '</div>' +
+                '<h1 id="firstHeading" class="firstHeading">Vehicle ID: ' + vehicle_ids[2*i+1] +'</h1>' +
+                '<div id="bodyContent">' +
+                '<p>License Plate: ' + license_plates[2*i+1] + '</p>' +
+                '<p>Date: ' + dates[2*i+1] + '</p>' +
+                '<p>Duration: ' + durations[2*i+1] + '</p>' +
```

Comment:
This seems like a bit hacky/messy. I propose a more elegant way to do this

Just pass a list of `IncidentReport`s to the template

```python
incident_reports = IncidentReport.query.all()
return render_template('main/map.html', incident_reports=incident_reports)
```

Then add the markers like this
```javascript
{% for report in incident_reports %}
var marker = new google.maps.Marker({
               //Use latitude and longitude values from incident report
               //to set position of marker
               position:{lat: '{{ report.location.latitude }}', lng: '{{ report.location.longitude }}' }
});
marker.setMap(map);
var contentString = '<div id="content">' +
'<div id="siteNotice">' +
'</div>' +
'<h1 id="firstHeading" class="firstHeading">Vehicle ID: ' + '{{ report. vehicle_id }}'
'</h1>' +
...
{% endfor %}
```

Note: For this to work, we will at least have to generate the markers in the maps.html file. This is a perfectly fine tradeoff because it will make code nicer in my opinion.